man/fi_peer.3.md

```diff
@@ -192,6 +192,120 @@ on the local node.
    creates a reference between the peer_cq object and its own cq.
 ```
 
+# PEER SRX
+
+The peer SRX defines a mechanism by which peer providers may share a common
+shared receive context.  This avoids the overhead of having separate receive
+queues, can eliminate memory copies, and ensures correct application level
+message ordering.
+
+The setup of a peer SRX is similar to the setup for a peer CQ outlined above.
+A fid_peer_srx object links the owner of the SRX with the peer provider.
+Peer SRXs are configured by the owner calling the peer's fi_srx_context()
+call with the FI_IMPORT_PEER flag set.  The context parameter passed to
+fi_srx_context() must be a struct fi_peer_srx_context.
+
+The data structures to support peer SRXs are defined as follows:
+
+```
+struct fid_peer_srx;
+
+/* Castable to dlist_entry */
+struct fi_peer_rx_entry {
+    struct fi_peer_rx_entry *next;
+    struct fi_peer_rx_entry *prev;
```

Comment:
why not use dlist_entry here directly?

Reply:
This is exposed with the public API.  dlist_entry is internal, and I don't think we want to define and expose a linked-list API from libfabric.

Reply:
That makes sense. Thanks.

Reply:
I debated whether to include this, since with the tcp provider it actually uses a single-linked list.  shm and rxm use a double-linked list.  But without something here, the owner would need to allocate a structure to track unexpected messages, but the peer already has this.  I went with this as the compromise.